packages/tre/src/plugins/core/index.ts

```diff
@@ -86,27 +90,84 @@ const BINDING_JSON_VERSION = "MINIFLARE_VERSION";
 const BINDING_SERVICE_USER = "MINIFLARE_USER";
 const BINDING_TEXT_CUSTOM_SERVICE = "MINIFLARE_CUSTOM_SERVICE";
 const BINDING_JSON_CF_BLOB = "CF_BLOB";
+const BINDING_DATA_LIVE_RELOAD_SCRIPT = "MINIFLARE_LIVE_RELOAD_SCRIPT";
+
+const LIVE_RELOAD_SCRIPT_PORT_PLACEHOLDER = "$PORT";
+const LIVE_RELOAD_SCRIPT_TEMPLATE = `<script defer type="application/javascript">
+(function () {
+  // Miniflare Live Reload
+  var url = new URL("/core/reload", location.origin);
+  url.protocol = url.protocol.replace("http", "ws");
+  url.port = ${LIVE_RELOAD_SCRIPT_PORT_PLACEHOLDER};
+  function reload() { location.reload(); }
+  function connect(reconnected) {
+    var ws = new WebSocket(url);
+    if (reconnected) ws.onopen = reload;
+    ws.onclose = function(e) {
+      e.code === 1012 ? reload() : e.code === 1000 || e.code === 1001 || setTimeout(connect, 1000, true);
+    }
+  }
+  connect();
+})();
```

Comment:
I'm slightly uneasy about this code—it's using quite old style JS, and seems a bit fragile. It would be nice to see an exponential backoff reconnection rather than a 1000ms one, as well? If possible, could it be made a function that interpolated the `LIVE_RELOAD_SCRIPT_PORT_PLACEHOLDER` variable as a function argument, rather than a constant?

Reply:
Or a variable binding